addons/web/static/tests/helpers/mock_server.js

```diff
@@ -110,7 +110,9 @@ var MockServer = Class.extend({
             console.log('%c[rpc] request ' + route, 'color: blue; font-weight: bold;', args);
             args = JSON.parse(JSON.stringify(args));
         }
-        return this._performRpc(route, args).then(function (result) {
+        var def = this._performRpc(route, args);
+        var abort = def.abort;
+        def = def.then(function (result) {
```

Comment:
`_performRpc` returns a `$.Deferred`, while `def.then` returns a `$.Deferred.promise()`.

This is important, because you cannot resolve or reject a promise, so it does not have the `resolve` or `reject` methods.

Reply:
How I think you could avoid this issue:

```
var def = this._performRpc(route, args);
var abort = def.abort;
var promise = def.then(...);
promise.abort = abort.bind(def);
return promise;
```

Reply:
But all tests return def without abort in test... and crash... also I used
``` var abort = def.abort.bind(def); ```